### PR TITLE
feat(replays): Delete videos on replay delete request

### PR DESCRIPTION
**Pull request by cmanallen**
If we did not delete this the videos would remain in storage until the retention-period expires.

Relates to: https://github.com/getsentry/sentry/issues/63255

### CODE REVIEW

File: src/sentry/replays/tasks.py [hunk 1]
```diff
@@ -55,15 +57,18 @@ def delete_replay_recording(project_id: int, replay_id: str) -> None:
...
         if segment.file_id:
             filestore_segments.append(segment)
         else:
             direct_storage_segments.append(segment)
 
     # Issue concurrent delete requests when interacting with a remote service provider.
-    if direct_storage_segments:
-        with cf.ThreadPoolExecutor(max_workers=100) as pool:
+    with cf.ThreadPoolExecutor(max_workers=100) as pool:
+        pool.map(_delete_if_exists, video_filenames)
```
[the remark this answers is not in the document]
Comment (cmanallen) [reply src/sentry/replays/tasks.py:70]: I don't think its necessary.  If you have a video it deletes.  If you don't it does nothing.

**Review by JoshFerge — APPROVED** [review]
there's no way to know if a replay segment is a video or a rrweb payload from segment metadata?

**Comment by cmanallen** [comment]
@JoshFerge I believe they download the RRWeb first and then fetch the video once they know its a mobile replay.

**Comment by JoshFerge** [comment]
> @JoshFerge I believe they download the RRWeb first and then fetch the video once they know its a mobile replay.

got it, makes sense.

**Pull request merged**